### PR TITLE
Translate ruby faq 1 (ko)

### PR DESCRIPTION
**Pull request by marocchino**
original text can found in:

https://github.com/ruby/www.ruby-lang.org/blob/master/en/documentation/faq/index.md
https://github.com/ruby/www.ruby-lang.org/blob/master/en/documentation/faq/1/index.md

### CODE REVIEW

File: ko/documentation/faq/1/index.md [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+있습니다.
+
+Ruby를 사용하여 서버를 작성하고 프로토타입을 실험하고 일상적인 프로그래밍 작업을
+수행할 수 있습니다. 완전히 통합된 객체 지향 언어인 Ruby는 확장성이 뛰어납니다.
+
+루비의 기능:
+
+* 간단한 구문,
+* 기본적인 객체지향 기능(클래스, 메소드, 객체 등),
+* 특별한 객체지향 기능(mixin, singleton method, renaming 등),
```
Comment (yous) [inline ko/documentation/faq/1/index.md:56]:
```suggestion
* 특별한 객체 지향 기능(믹스인, 싱글턴 메서드, 리네이밍 등),
```

File: ko/documentation/faq/1/index.md [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+~~~
+Bert:    8
+Cookie: 11
+Elmo:    4
+Ernie:   8
+Zoe:     7
+~~~
+
+이 코드는 정규식을 사용하여 입력 파일에서 연속적인 라인을 구문 분석하여 각 일치
```
Comment (yous) [inline ko/documentation/faq/1/index.md:101]:
```suggestion
이 코드는 정규식을 사용하여 입력 파일에서 연속적인 줄을 구문 분석하여 각 일치
```

File: ko/documentation/faq/1/index.md [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+항목에 대해 새 `Person` 개체를 만들고 배열 `people`의 끝으로 푸시합니다.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:102]:
```suggestion
항목에 대해 새 `Person` 객체를 만들고 배열 `people`의 맨 뒤에 푸시합니다.
```

File: ko/documentation/faq/1/index.md [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+people = Array.new
+
+File.foreach("ages") do |line|
+  people << Person.new($1, $2) if line =~ /(.*):\s+(\d+)/
+end
+
+people  # => [Bert (8), Cookie (11), Elmo (4), Ernie (8), Zoe (7)]
+~~~
+
+이제 사람의 나이를 기준으로 결과를 정렬해 보겠습니다. 여러 가지가 방법으로 이를
```
Comment (yous) [inline ko/documentation/faq/1/index.md:114]:
```suggestion
이제 사람의 나이를 기준으로 결과를 정렬해 보겠습니다. 여러 가지 방법으로 이를
```

File: ko/documentation/faq/1/index.md [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+people.sort  # => [Elmo (4), Zoe (7), Bert (8), Ernie (8), Cookie (11)]
+~~~
+
+### 이름이 "Ruby"인 이유는 무엇인가요?
+
+Perl의 영향을 받은 Matz는 새로운 언어에 보석 이름을 사용하고 싶었고 동료의
+탄생석을 따서 Ruby라는 이름을 지었습니다.
+
+나중에 그는 Ruby가 여러 상황에서 Perl 바로 다음에 오는 것을 깨달았습니다.
+탄생석에서 Perl은 6월, Ruby는 7월입니다. 글자크기 측정시 Perl은 5pt, Ruby는
```
Comment (yous) [inline ko/documentation/faq/1/index.md:140]:
```suggestion
탄생석에서 Perl은 6월, Ruby는 7월입니다. 폰트 크기 측정 시 pearl은 5pt, ruby는
```

File: ko/documentation/faq/1/index.md [hunk 6]
```diff
@@ -0,0 +1,319 @@
...
+5.5pt 입니다. 그는 Ruby가 Perl보다 새로운(그리고 바라건대 더 나은) 프로그래밍
```
Comment (yous) [inline ko/documentation/faq/1/index.md:141]:
```suggestion
5.5pt입니다. 그는 Ruby가 Perl보다 새로운(그리고 바라건대 더 나은) 프로그래밍
```

File: ko/documentation/faq/1/index.md [hunk 7]
```diff
@@ -0,0 +1,319 @@
...
+언어의 좋은 이름이라고 생각했습니다.
+
+(1999년 6월 11일 [\[ruby-talk:00394\]][ruby-talk:00394]에서 Matz의 설명을
+기반으로 합니다.)
+
+[ruby-talk:00394]: https://blade.ruby-lang.org/ruby-talk/394
+
+### Ruby의 역사는?
+
+다음은 1999년 6월 4일에 [\[ruby-talk:00382\]][ruby-talk:00382]에 Matz가 작성한
```
Comment (yous) [inline ko/documentation/faq/1/index.md:151]:
```suggestion
다음은 1999년 6월 4일 [\[ruby-talk:00382\]][ruby-talk:00382]에 Matz가 작성한
```

File: ko/documentation/faq/1/index.md [hunk 8]
```diff
@@ -0,0 +1,319 @@
...
+게시물의 요약입니다. (Ruby의 생일은 [\[ruby-list:15977\]][ruby-list:15977]에서
```
Comment (yous) [inline ko/documentation/faq/1/index.md:152]:
```suggestion
게시물을 요약한 것입니다. (Ruby의 생일은 [\[ruby-list:15977\]][ruby-list:15977]에서
```

File: ko/documentation/faq/1/index.md [hunk 9]
```diff
@@ -0,0 +1,319 @@
...
+수정되었습니다.)
+
+> Ruby는 1993년 2월 24일에 태어났습니다. 동료와 객체 지향 스크립팅 언어의
+> 가능성에 대해 이야기하고 있었습니다. 나는 Perl(Perl5가 아닌 Perl4)을 알고
+> 있었지만 장난감 언어의 냄새가 나기 때문에 별로 좋아하지 않았습니다(여전히
+> 그렇습니다). 객체 지향 스크립팅 언어는 매우 유망해 보였습니다.
+
+> 저는 그때 Python을 알았습니다. 그러나 나는 그것이 진정한 객체 지향 언어라고
+> 생각하지 않았기 때문에 그것을 좋아하지 않았습니다. OO 기능은 언어에 대한 추가
```
Comment (yous) [inline ko/documentation/faq/1/index.md:161]:
```suggestion
> 생각하지 않았기 때문에 그것을 좋아하지 않았습니다. 객체 지향 기능은 언어에 대한 추가
```

File: ko/documentation/faq/1/index.md [hunk 10]
```diff
@@ -0,0 +1,319 @@
...
+> 기능으로 나타났습니다. 15년 동안 언어 매니아이자 OO 팬으로서 저는 진정한 객체
```
Comment (yous) [inline ko/documentation/faq/1/index.md:162]:
```suggestion
> 기능으로 나타났습니다. 15년 동안 언어 마니아이자 객체 지향 팬으로서 저는 진정한 객체
```

File: ko/documentation/faq/1/index.md [hunk 11]
```diff
@@ -0,0 +1,319 @@
...
+> 지향의 사용하기 쉬운 스크립팅 언어를 정말로 원했습니다. 찾아보았지만 찾을 수
+> 없었습니다.
+
+> 그래서 만들기로 했습니다. 인터프리터를 가동시키는 데 몇 달이 걸렸습니다.
+> 반복자, 예외 처리, 가비지 콜렉터 같은 내가 좋아하는 기능을 내 언어에
```
Comment (yous) [inline ko/documentation/faq/1/index.md:167]:
```suggestion
> 이터레이터, 예외 처리, 가비지 컬렉터 같은 내가 좋아하는 기능을 내 언어에
```

File: ko/documentation/faq/1/index.md [hunk 12]
```diff
@@ -0,0 +1,319 @@
...
+
+> 그 이후로 활발한 메일링 리스트가 만들어지고 웹 페이지가 형성되었습니다.
+
+[ruby-talk:00382]: https://blade.ruby-lang.org/ruby-talk/382
+[ruby-list:15977]: https://blade.ruby-lang.org/ruby-list/15977
+
+### Ruby 홈 페이지는 어디에 있나요?
+
+공식 Ruby 홈 페이지는 [www.ruby-lang.org](https://www.ruby-lang.org)입니다.
+영어 및 일본어 버전 외에도 다양한 다른 언어로의 번역이 있습니다.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:181]:
```suggestion
영어 및 일본어 버전 외에도 다양한 언어로 번역되어 있습니다.
```

File: ko/documentation/faq/1/index.md [hunk 13]
```diff
@@ -0,0 +1,319 @@
...
+
+Ruby 정보를 찾기 위한 좋은 출발점은 [문서](/ko/documentation/) 및
+[커뮤니티](/ko/community/) 페이지입니다.
+
+### Ruby 뉴스그룹이 있나요?
+
+comp.lang.ruby는 2000년 5월에 설립되었습니다.
+([Conrad Schneiker](mailto:schneiker@jump.net)의 노력에 감사합니다)
```
Comment (yous) [inline ko/documentation/faq/1/index.md:189]:
```suggestion
([Conrad Schneiker](mailto:schneiker@jump.net)의 노력에 감사드립니다.)
```

File: ko/documentation/faq/1/index.md [hunk 14]
```diff
@@ -0,0 +1,319 @@
...
+
+### Ruby 메일링 리스트가 있나요?
+
+Ruby에 대해 이야기하는 여러 메일링 리스트가 있습니다.
+자세한 내용은 [메일링 리스트](/ko/community/mailing-lists/) 페이지를 참조하세요.
+
+[https://ml.ruby-lang.org/mailman3/hyperkitty/list/ruby-talk@ml.ruby-lang.org/](https://ml.ruby-lang)를
```
Comment (yous) [inline ko/documentation/faq/1/index.md:196]:
```suggestion
<https://ml.ruby-lang.org/mailman3/hyperkitty/list/ruby-talk@ml.ruby-lang.org/>를
```

File: ko/documentation/faq/1/index.md [hunk 15]
```diff
@@ -0,0 +1,319 @@
...
+사용하여 메일링 리스트 아카이브를 검색할 수 있습니다. (이것은 ruby-talk 목록의
+URL이며, 다른 항목에 필요한 경우 변환됩니다.)
```
Comment (yous) [inline ko/documentation/faq/1/index.md:198]:
```suggestion
URL이며, 다른 메일링 리스트를 원하면 변경해서 사용하세요.)
```

File: ko/documentation/faq/1/index.md [hunk 16]
```diff
@@ -0,0 +1,319 @@
...
+comp.lang.ruby는 2000년 5월에 설립되었습니다.
+([Conrad Schneiker](mailto:schneiker@jump.net)의 노력에 감사합니다)
+
+### Ruby 메일링 리스트가 있나요?
+
+Ruby에 대해 이야기하는 여러 메일링 리스트가 있습니다.
+자세한 내용은 [메일링 리스트](/ko/community/mailing-lists/) 페이지를 참조하세요.
+
+[https://ml.ruby-lang.org/mailman3/hyperkitty/list/ruby-talk@ml.ruby-lang.org/](https://ml.ruby-lang)를
+사용하여 메일링 리스트 아카이브를 검색할 수 있습니다. (이것은 ruby-talk 목록의
```
Comment (yous) [inline ko/documentation/faq/1/index.md:197]:
```suggestion
사용하여 메일링 리스트 아카이브를 검색할 수 있습니다. (이것은 ruby-talk 메일링 리스트의
```

File: ko/documentation/faq/1/index.md [hunk 17]
```diff
@@ -0,0 +1,319 @@
...
+~~~
+
+### "Ruby"와 "ruby" 중 어느 것이 맞습니까?
+
+공식적으로는 언어를 "Ruby"라고 합니다. 대부분의 시스템에서는 `ruby` 명령을
+사용하여 호출됩니다. "Ruby" 대신 "ruby"를 사용해도 괜찮습니다.
+
+언어 이름으로 "RUBY"를 사용하지 마십시오.
+
+원래 또는 역사적으로 "ruby"라고 불렸습니다.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:222]:
```suggestion
처음엔, 역사적으로도, "ruby"라고 불렸습니다.
```

File: ko/documentation/faq/1/index.md [hunk 18]
```diff
@@ -0,0 +1,319 @@
...
+
+{% include faq-out-of-date.html %}
+
+* Ruby 프로그래밍: 실용적인 프로그래머 가이드,
+  (곡괭이 책) David Thomas와 Andrew Hunt 저: ISBN 0-20171-089-7,
+  Addison-Wesley, October 2000.
+
+* Matz 외의 일본어 Ruby 참고서. 일본에서는 ASCII에서 출판되었습니다(ISBN
+  4-7561-3254-5). 영어 번역본인 "The Ruby Programming Language"는 O'Reilly &
+  Associates에서 구할 수 있습니다. (ISBN 978-0596516178)
```
Comment (yous) [inline ko/documentation/faq/1/index.md:234]:
```suggestion
  Associates에서 구할 수 있습니다(ISBN 978-0596516178).
```

File: ko/documentation/faq/1/index.md [hunk 19]
```diff
@@ -0,0 +1,319 @@
...
+
+* 일본어 "Ruby Pocket Reference"는 O'Reilly Japan에서 출판되었습니다(ISBN
+  4-87311-023-8). 번역본을 보고 싶다면 미국에 있는 O'Reilly에게 알려주십시오.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:237]:
```suggestion
  4-87311-023-8). 번역본을 보고 싶다면 미국에 있는 O'Reilly에게 알려주세요.
```

File: ko/documentation/faq/1/index.md [hunk 20]
```diff
@@ -0,0 +1,319 @@
...
+
+원래 또는 역사적으로 "ruby"라고 불렸습니다.
+
+### Ruby 책이 있나요?
+
+{% include faq-out-of-date.html %}
+
+* Ruby 프로그래밍: 실용적인 프로그래머 가이드,
+  (곡괭이 책) David Thomas와 Andrew Hunt 저: ISBN 0-20171-089-7,
+  Addison-Wesley, October 2000.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:230]:
```suggestion
  Addison-Wesley, 2000년 10월.
```

File: ko/documentation/faq/1/index.md [hunk 21]
```diff
@@ -0,0 +1,319 @@
...
+
+* Matz 외의 일본어 Ruby 참고서. 일본에서는 ASCII에서 출판되었습니다(ISBN
+  4-7561-3254-5). 영어 번역본인 "The Ruby Programming Language"는 O'Reilly &
+  Associates에서 구할 수 있습니다. (ISBN 978-0596516178)
+
+* 일본어 "Ruby Pocket Reference"는 O'Reilly Japan에서 출판되었습니다(ISBN
+  4-87311-023-8). 번역본을 보고 싶다면 미국에 있는 O'Reilly에게 알려주십시오.
+
+* 또한 Jeffrey Friedl의 "정규 표현식 마스터링"(Hip Owl Book): O'Reilly &
+  Associates의 ISBN 1-56592-257-3은 다양한 프로그래밍 언어에서 정규 표현식의
```
Comment (yous) [inline ko/documentation/faq/1/index.md:240]:
```suggestion
  Associates, ISBN 1-56592-257-3은 다양한 프로그래밍 언어에서 정규 표현식의
```

File: ko/documentation/faq/1/index.md [hunk 22]
```diff
@@ -0,0 +1,319 @@
...
+  기술 및 구현을 다루는 참조 작업입니다. 대부분 Ruby 정규 표현식과 관련성이
```
Comment (yous) [inline ko/documentation/faq/1/index.md:241]:
```suggestion
  기술 및 구현을 다루는 참고서입니다. 대부분 Ruby 정규 표현식과 관련성이
```

File: ko/documentation/faq/1/index.md [hunk 23]
```diff
@@ -0,0 +1,319 @@
...
+  XEmacs의 일부 버전에서는 `.emacs` 파일에 `(load "font-lock")`을 추가해야
+  `ruby-mode.el`이 사용 중인 구문 강조 패키지를 감지할 수 있습니다.
+* [Vim](http://www.vim.org/): Vim 5.7 이상에는 런타임 패키지의 표준으로 Ruby
+  구문 파일이 있습니다. 이전 버전의 경우 Ruby용 구문 파일은
+  [http://www.xs4all.nl/~hipster/lib/ruby/ruby.vim](http://www.xs4all.nl/~hipster/lib/ruby/ruby.vim)에서 찾을 수 있습니다.
+* [Jedit](http://jedit.sourceforge.net/): Java로 작성된 휴대용 편집기로 Ruby를
+  지원합니다.
+* [Nedit](http://www.nedit.org): Eric Santonacci는 Nedit에 대한 Ruby 지원을
+  작성했습니다.
+  [ftp://ftp.talc.fr/pub/ruby/ruby.nedit-0.1.tar.gz](ftp://ftp.talc.fr/pub/ruby/ruby.nedit-0.1.tar.gz).
```
Comment (yous) [inline ko/documentation/faq/1/index.md:259]:
```suggestion
  [ftp://ftp.talc.fr/pub/ruby/ruby.nedit-0.1.tar.gz](ftp://ftp.talc.fr/pub/ruby/ruby.nedit-0.1.tar.gz)에서 사용 가능합니다.
```

File: ko/documentation/faq/1/index.md [hunk 24]
```diff
@@ -0,0 +1,319 @@
...
+* Barry Shultz는 TextPad용 Ruby 정의 파일을 작성했습니다.
+  [https://www.textpad.com/add-ons/synn2t.html](https://www.textpad.com/add-ons/synn2t.html).
```
Comment (yous) [inline ko/documentation/faq/1/index.md:261]:
```suggestion
  [https://www.textpad.com/add-ons/synn2t.html](https://www.textpad.com/add-ons/synn2t.html)에서 사용 가능합니다.
```

File: ko/documentation/faq/1/index.md [hunk 25]
```diff
@@ -0,0 +1,319 @@
...
+
+### Ruby 코드의 실행 결과를 주석으로 추가하려면 어떻게 해야하나요?
+
+{% include faq-out-of-date.html %}
+
+사람들은 일반적으로 각 명령문 실행 결과를 해당 명령문에 첨부된 주석으로 표시하여
+Ruby 코드에 주석을 답니다. 예를 들어 다음 코드에서는 할당이 문자열 "Billy Bob"을
+생성한 다음 일부 하위 문자열을 추출한 결과를 보여줍니다.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:269]:
```suggestion
생성한 다음 일부 부분 문자열을 추출한 결과를 보여줍니다.
```

File: ko/documentation/faq/1/index.md [hunk 26]
```diff
@@ -0,0 +1,319 @@
...
+  "Pipe the region through Ruby's xmp utility and replace the region with
+   the result."
+  (interactive "r")
+  (shell-command-on-region reg-start reg-end
+                           "ruby -r xmp -n -e 'xmp($_, \"%l\t\t# %r\n\")'"
+                           t))
+(global-set-key [(meta f10)] 'ruby-xmp-region)
+~~~
+
+Vim 사용자는 매핑을 사용할 수 있습니다(hipster님 감사합니다):
```
Comment (yous) [inline ko/documentation/faq/1/index.md:291]:
```suggestion
Vim 사용자는 매핑을 사용할 수 있습니다(hipster님 감사합니다).
```

File: ko/documentation/faq/1/index.md [hunk 27]
```diff
@@ -0,0 +1,319 @@
...
+~~~
+
+두 경우 모두 코드 영역을 강조 표시하고 Meta-F10을 눌러 주석을 추가합니다.
+
+### Ruby 매뉴얼을 읽어도 이해가 안 돼요!
+
+{% include faq-out-of-date.html %}
+
+Ruby의 구문은 Ruby 1.0 이후 상당히 안정적이지만 때때로 새로운 기능이 추가됩니다.
+따라서 책과 온라인 설명서가 뒤쳐질 수 있습니다.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:304]:
```suggestion
따라서 책과 온라인 설명서가 뒤처질 수 있습니다.
```

File: ko/documentation/faq/1/index.md [hunk 28]
```diff
@@ -0,0 +1,319 @@
...
+
+문제가 있는 경우 메일링 리스트([메일링 리스트
+페이지](/ko/community/mailing-lists/) 참조)에서 언제든지 질문하십시오.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:307]:
```suggestion
문제가 있는 경우 메일링 리스트([메일링 리스트 페이지](/ko/community/mailing-lists/) 참조)에서
언제든지 질문하십시오.
```

File: ko/documentation/faq/1/index.md [hunk 29]
```diff
@@ -0,0 +1,319 @@
...
+일반적으로 언어의 작성자인 Matz 자신, 다른 전문가 및 자신과 유사한 문제를 해결한
```
Comment (yous) [inline ko/documentation/faq/1/index.md:308]:
```suggestion
일반적으로 언어의 작성자인 Matz, 다른 전문가, 자신과 유사한 문제를 해결한
```

File: ko/documentation/faq/1/index.md [hunk 30]
```diff
@@ -0,0 +1,319 @@
...
+사람들로부터 적절한 답변을 얻을 수 있습니다.
+
+문제가 있는 소스 코드와 함께 `ruby -v`의 출력을 포함하십시오.
+
+[`irb`](en/documentation/faq/10/#irb)를 사용하는 데 문제가 있는 경우 몇 가지
```
Comment (yous) [inline ko/documentation/faq/1/index.md:313]:
```suggestion
[`irb`](/en/documentation/faq/10/#irb)를 사용하는 데 문제가 있는 경우 몇 가지
```

File: ko/documentation/faq/1/index.md [hunk 31]
```diff
@@ -0,0 +1,319 @@
...
+제한 사항이 있다는 점에 유의하세요. `irb --single-irb`를 사용하거나 `ruby`
+명령을 직접 사용하여 스크립트를 사용해 보십시오.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:315]:
```suggestion
명령을 직접 사용하여 스크립트를 사용해 보세요.
```

File: ko/documentation/faq/1/index.md [hunk 32]
```diff
@@ -0,0 +1,319 @@
...
+
+Ruby의 구문은 Ruby 1.0 이후 상당히 안정적이지만 때때로 새로운 기능이 추가됩니다.
+따라서 책과 온라인 설명서가 뒤쳐질 수 있습니다.
+
+문제가 있는 경우 메일링 리스트([메일링 리스트
+페이지](/ko/community/mailing-lists/) 참조)에서 언제든지 질문하십시오.
+일반적으로 언어의 작성자인 Matz 자신, 다른 전문가 및 자신과 유사한 문제를 해결한
+사람들로부터 적절한 답변을 얻을 수 있습니다.
+
+문제가 있는 소스 코드와 함께 `ruby -v`의 출력을 포함하십시오.
```
Comment (yous) [inline ko/documentation/faq/1/index.md:311]:
```suggestion
문제가 있는 소스 코드와 함께 `ruby -v`의 출력을 첨부하세요.
```

File: ko/documentation/faq/index.md [hunk 33]
```diff
@@ -0,0 +1,62 @@
+---
+layout: page
+title: "공식 Ruby FAQ"
+lang: ko
+
+header: |
+  <div class="multi-page">
+    <strong>콘텐츠</strong>
+    <span class="separator"> | </span>
+    <a href="1/" title="Part 1">1</a>
```
[the remark this answers is not in the document]
Comment (marocchino) [reply ko/documentation/faq/index.md]: apply in [9a1650f](https://github.com/ruby/www.ruby-lang.org/pull/2998/commits/9a1650f82c2a5406986fd165fdb9e45711a0359f)

**Review by yous — APPROVED** [review]

**Pull request merged**